tools/hermetic_grunt/Dockerfile

```diff
@@ -0,0 +1,54 @@
+# To build and run the hermetic grunt:
+# docker build -f tools/hermetic_grunt/Dockerfile -t uproxy-grunt . && \
+# docker run --rm -v $(pwd):/root/repository_root -it uproxy-grunt
+#
+# You have to run from the repository root. The output is written to the out/
+# directory.
+
+FROM ubuntu
+
+LABEL description="Docker container for hermetic grunt runs"
+
+###############################################################################
+# Install prerequisite global tools
+###############################################################################
+
+RUN apt-get update
+
+RUN apt-get install -qqy git python build-essential
+
+# Add a fresher repository for nodejs
+RUN apt-get install curl && \
+  curl -sL https://deb.nodesource.com/setup_4.x | bash - && \
+  apt-get install -qqy nodejs && \
+  apt-get purge -y --auto-remove curl
```

Comment:
I've also removed the need for explicitly installing curl. Docker's ADD can fetch things from the internet.

In any case, I think some of the tools install curl, so it's in the image.

On using it to interact with the sample apps, I think it doesn't apply, since you wouldn't be doing that from inside the container?